### PR TITLE
Test case for ComposedId Entity with Lazy Property is not proxified

### PR DESCRIPTION
**Pull request by MarkPerryBV**
#1439

### CODE REVIEW

File: src/NHibernate.Test/NHSpecificTest/GH1439/Entity.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		{
+			return (Id + "|" + Name).GetHashCode();
+		}
+
+		public override bool Equals(object obj)
+		{
+			if (obj == null)
+				return false;
+
+			return obj.GetHashCode() == GetHashCode();
```
Comment (fredericDelaporte) [inline src/NHibernate.Test/NHSpecificTest/GH1439/Entity.cs]: Even if it is only a test, please use a correct implementation. Hashcode are not guaranteed unique per entity. Hashcode collisions occur and create bugs. Testing hashcode equality can only serves as a shortcut for early exit if they are unequal. If they are equal, the entities may still be different. So for such a simple case, better just test directly that both `Id` and `Name` are equals.

File: src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
+﻿using NHibernate.Test.NHSpecificTest.GH0000;
```
Comment (fredericDelaporte) [inline src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs]: Please do not keep a dependency on the "template" bug. (It seems removable without any other changes here by the way.)

File: src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			{
+				session.Delete("from System.Object");
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		[Test]
+		public void YourTestName()
```
Comment (fredericDelaporte) [inline src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs]: This should be renamed to something more meaningful. Like `LazyPropertyShouldNotBeNullified`.

File: src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
+﻿using NHibernate.Test.NHSpecificTest.GH0000;
+
+namespace NHibernate.Test.NHSpecificTest.GH1439
```
Comment (fredericDelaporte) [inline src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs]: There are a bunch of using lacking here. It does not compile.

File: src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+			using (var tran = session.BeginTransaction())
+			{
+				var result = (from e in session.Query<CompositeEntity>()
+							 where e.Name == "Bob"
+							 select e).ToList();
+				session.Flush();
+				tran.Commit();
+				Assert.AreEqual(1, result.ToList().Count);
+			}
+		}
```
Comment (fredericDelaporte) [inline src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs]: For asserting the bug, you may:
- Enable statistics on the session factory for checking the update count. (See by example [here](https://github.com/nhibernate/nhibernate-core/pull/1414/files#diff-2fd7f1e84082ebfc50ebd804dd007385R93), it does it for an insert case.)
- Reload the entity at the end of the test, and assert the value of the lazy property.

File: src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		{
+			using (ISession session = OpenSession())
+			using (var tran = session.BeginTransaction())
+			{
+				var result = (from e in session.Query<CompositeEntity>()
+							 where e.Name == "Bob"
+							 select e).ToList();
+				session.Flush();
+				tran.Commit();
+				Assert.AreEqual(1, result.ToList().Count);
```
Comment (fredericDelaporte) [inline src/NHibernate.Test/NHSpecificTest/GH1439/FixtureByCode.cs]: Redundant `ToList`.

**Comment by hazzik** [comment]
Need to regenerate async tests.

**Comment by fredericDelaporte** [comment]
I have added a test with the id mapped as a component to demonstrate the issue is specific to using a composite-id without a component.

**Comment by fredericDelaporte** [comment]
The failing case, using a composite-id without a component, is now marked as known bug.

**Pull request merged**